test/functional/helpers.lua

```diff
@@ -577,8 +577,16 @@ function module.assert_alive()
 end
 
 local function do_rmdir(path)
-  if lfs.attributes(path, 'mode') ~= 'directory' then
-    return  -- Don't complain.
+  local mode, errmsg, errcode = lfs.attributes(path, 'mode')
```

Comment:
For reference: older versions of lfs (before 1.7.0) did not have `errcode` retval, then the `string.format()` call below will fail.  

- https://github.com/keplerproject/luafilesystem/commit/b37e88b3d6128b85586231200c6d1a2f05ecd0d6
- https://github.com/keplerproject/luafilesystem/commit/81e5b165bf324689777b967143c751786c405ce4

Also `errcode` is system-dependent, so the `errcode == 2` check below is best-effort I guess.